synapse/rest/client/v2_alpha/sendtodevice.py

```diff
@@ -52,10 +59,11 @@ def on_PUT(self, request, message_type, txn_id):
 
     async def _put(self, request, message_type, txn_id):
         requester = await self.auth.get_user_by_req(request, allow_guest=True)
+        sender_user_id = requester.user.to_string()
 
-        content = parse_json_object_from_request(request)
+        self._ratelimiter.ratelimit(sender_user_id)
```

Comment:
Does it make sense to limit *all* requests on this endpoint or do we want to limit in `send_device_message` as #8675 did?

Downsides of doing it in the loop in `send_device_message` is that we could *partially* handle a request and then raise a limit exceeded error, which seems bad. 😢 

It also seems like we wouldn't want to rate limit key share requests to yourself?

Additionally, do we want different limits for local vs. remote as some of the other rate limiters have?

Reply:
Yeah, I think this does have to look at if its a key share request. We do a lot of to device messages that are really not OK to drop without breaking encryption sessions.

Reply:
Should we then *drop* those requests or raise a rate limit error?

Reply:
Sorry, I'm not following?